Sources/MongoKitten/AggregateStage.swift

```diff
@@ -1,227 +1,289 @@
-        assert(n > 0)
-        
-        return AggregateBuilderStage(document: [
-            "$sample": ["size": n]
-        ])
+    public init(from: String, localField: FieldPath, foreignField: FieldPath, as newName: FieldPath) {
+        self.stage = [
+            "$lookup":
+                [
+                    "from": from,
+                    "localField": localField.string,
+                    "foreignField": foreignField.string,
+                    "as": newName.string
+                ]
+        ]
     }
     
-	/// The `lookup` aggregation performs a join from another collection in the same database. This aggregation will add a new array to
-	/// your document including the matching documents.
-	///
-	/// # MongoDB-Documentation:
-	/// [Link to the MongoDB-Documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/)
-	///
-	/// # Example:
-	/// There are two collections, named `users` and `userCategories`. In the `users` collection there is a reference to the _id
-	/// of the `userCategories`, because every user belongs to a category.
-	///
-	/// If you now want to aggregate all users and the corresponding user category, you can use the `$lookup` like this:
-	///
-	/// ```
-	/// let pipeline = userCollection.aggregate([
-	///     .lookup(from: "userCategories", "localField": "categoryID", "foreignField": "_id", newName: "userCategory")
-	/// ])
-	///
-	/// pipeline.execute().whenComplete { result in
-	///    ...
-	/// }
-	/// ```
-	///
-	/// # Hint:
-	/// Because the matched documents will be inserted as an array no matter if there is only one item or more, you may want to unwind the joined documents:
-	///
-	/// ```
-	/// let pipeline = myCollection.aggregate([
-	///     .lookup(from: ..., newName: "newName"),
-	///     .unwind(fieldPath: "$newName")
-	/// ])
-	/// ```
-	///
-	/// - Parameters:
-	///   - from: the foreign collection, where the documents will be looked up
-	///   - localField: the name of the field in the input collection that shall match the `foreignField` in the `from` collection
-	///   - foreignField: the name of the field in the `fromCollection` that shall match the `localField` in the input collection
-	///   - newName: the collecting matches will be inserted as an array to the input collection, named as `newName`
-	/// - Returns: an `AggregateBuilderStage`
-    public static func lookup(
-        from: String,
-        localField: String,
-        foreignField: String,
-        as newName: String
-    ) -> AggregateBuilderStage {
-        return AggregateBuilderStage(document: [
-            "$lookup": [
-                "from": from,
-                "localField": localField,
-                "foreignField": foreignField,
-                "as": newName
-            ]
-        ])
+    public init(from: String, usedPipelineVariables: Document = [:], @AggregateBuilder pipeline: () -> [AggregateBuilderStage], as newField: FieldPath) {
```

Comment:
These pipeline variables should be optional still, or make a new `init` where the pipeline variables are not accepted as an argument.